cargo-miri/bin.rs

```diff
@@ -112,40 +112,60 @@ fn has_arg_flag(name: &str) -> bool {
     args.any(|val| val == name)
 }
 
-/// Yields all values of command line flag `name`.
-struct ArgFlagValueIter<'a> {
-    args: TakeWhile<env::Args, fn(&String) -> bool>,
+/// Yields all values of command line flag `name` as `Ok(arg)`, and all other arguments except
+/// the flag as `Err(arg)`.
+struct ArgFlagValueWithOtherArgsIter<'a, I> {
```

Comment:
I think this should have a shorter name, but I can't think of any now...
Also, it seems a bit opaque to use `Result` here.

Reply:
> (actually I don't know how to name it either)

The proper name would be `Either` :joy: 
(that's the Haskell name for `Result`, not tied to error handling)

But I agree it doesn't seem worth it.

Reply:
About the struct name, what about `ArgSplitFlagValue`?